ASB resumes unfinished swaps after startup (#376)
comit-network/xmr-btc-swap: CHANGELOG.md

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](

),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Added

- A changelog file.
- The `asb` resumes unfinished swaps upon startup.
  This is a breaking changed because we now save the peer-id of the counterparty in the database.

>   It is recommended to reset the database before running the upgraded version of the `asb` because any unfinished swaps will be tried for resume but will fail without the counterparty peer-id being present.

Reviewer: Also, can we please squash this commit into the one with the actual feature?

>   It is recommended to reset the database before running the upgraded version of the `asb` because any unfinished swaps will be tried for resume but will fail without the counterparty peer-id being present.

Author: > This seems to be an unnecessary burden for the user? I'd say let's just skip the swaps where we don't have a PeerId.

Can do - and print a warning in the logs. Otherwise the user does not know that "unsupported" swaps are being skipped.

## [Unreleased]

### Changed

> - The `resume` command of the `swap` CLI to no longer require the `--seller-peer-id` parameter.

Reviewer: ```suggestion
- The `resume` command of the `swap` CLI no longer require the `--seller-peer-id` parameter.
```

### Added

- A changelog file.
- Automatic resume of unfinished swaps for the `asb` upon startup.

>   Unfinished swaps that were triggered prior to this change will be skipped by resume.

Reviewer: ```suggestion
  Unfinished swaps from earlier versions will be skipped.
```

- The `asb` resumes unfinished swaps upon startup.
  This is a breaking changed because we now save the peer-id of the counterparty in the database.
  It is recommended to reset the database before running the upgraded version of the `asb` because any unfinished swaps will be tried for resume but will fail without the counterparty peer-id being present.

> - The `swap` CLI saves the seller-peer-id into the database to be re-used for `resume`, so the `--seller-peer-id` parameter for resume was removed.

Reviewer: I know it is not always easy to do but I think it would be worth it to strive for a wording where these entries use the "heading" as their first word.

Like:

**Added** _a changelog file_.

Additionally, I think we should word it in a "user-first" mentality. What is important to the user is that the `resume` command **changed**, not that we added stuff to the database.

My suggestion would be to move this to the **Changed** section and say:

### Changed

- The `resume` command of the `swap` CLI to no longer require the `--seller-peer-id` parameter. This information is now saved in the database.
